server/shadow/shadow_client.c

```diff
@@ -30,10 +30,62 @@
 
 #include <freerdp/log.h>
 
+#include "../../libfreerdp/core/server.h"
 #include "shadow.h"
 
 #define TAG CLIENT_TAG("shadow")
 
+struct _SHADOW_GFX_STATUS
+{
+	BOOL gfxOpened;
+	BOOL gfxSurfaceCreated;
+};
+typedef struct _SHADOW_GFX_STATUS SHADOW_GFX_STATUS;
+
+static void shadow_client_rdpgfx_new_surface(rdpShadowClient *client)
+{
```

Comment:
Reply your comments both this and the next one :)

This is tricky, as I haven't figure out what to do if graphic commands fails. Currently I have to just ignore it (this function, new_surface should be handled, if fail the surface need to be recreated next time. But reset and release, just ignore).

But I agree to have them with return value, just ignore when calling.